File: yht0827/2월/2월 8일/H-Index.java
import java.util.*;

class Solution {
	public int solution(int[] citations) {
		int answer = 0;
		int n = citations.length;

		Arrays.sort(citations);

		for (int i = 0; i < n; i++) {
			int h = n - i;

			if (citations[i] >= h) {
				answer = h;
				break;
			}
		}

		return answer;
	}
}

File: zuzubibi/2월/2월 8일/H-index.py
def solution(citations):
    answer = 0

    n = len(citations)
    citations.sort()

    for i in range(n):
        hIndex = n-i

        if citations[i] >= hIndex:
            answer= hIndex
            break

    return answer

# 3
print(solution([3,0,6,1,5]))